smartsim/_core/utils/serialize.py

```diff
@@ -0,0 +1,216 @@
+) -> None:
+    manifest_dir = Path(experiment.exp_path) / ".smartsim/manifest"
+    manifest_dir.mkdir(parents=True, exist_ok=True)
+    manifest_file = manifest_dir / "manifest.json"
+
+    # FIXME: Did we decide on if this should be uuid/literal count of the
+    #        runs/time since epoch/etc.? Whatever it is, it probably should not
+    #        be secs since epoch as this could lead to conflicts if many
+    #        non-blocking runs are made in quick succession.
+    run_id = int(time.time())
+
+    new_run = {
+        "run_id": run_id,
+        "model": [
+            _dictify_model(
+                model,
+                *telemetry_metadata,
+                manifest_dir / f"{experiment.name}/{run_id}/model",
+            )
+            for model, telemetry_metadata in manifest.models
+        ],
+        "orchestrator": [
+            _dictify_db(
+                db, nodes_info, manifest_dir / f"{experiment.name}/{run_id}/database"
+            )
+            for db, nodes_info in manifest.database
+        ],
+        "ensemble": [
+            _dictify_ensemble(
+                ens, member_info, manifest_dir / f"{experiment.name}/{run_id}/ensemble"
+            )
+            for ens, member_info in manifest.ensembles
+        ],
+    }
+    try:
+        with open(manifest_file, "r") as fd:
+            manifest_dict = json.load(fd)
+    except (FileNotFoundError, json.JSONDecodeError):
+        manifest_dict = {
+            "experiment": _dictify_experiment(experiment),
+            "runs": [new_run],
+        }
+    else:
+        manifest_dict["runs"].append(new_run)
+    finally:
+        with open(manifest_file, "w") as fd:
+            json.dump(manifest_dict, fd, indent=2)
+
+
+def _dictify_experiment(exp: Experiment) -> t.Dict[str, str]:
+    return {
+        "name": exp.name,
+        "path": exp.exp_path,
+        "launcher": exp._launcher,
+    }
+
+
+def _dictify_model(
+    model: Model,
+    step_id: t.Optional[str],
+    task_id: t.Optional[str],
+    managed: t.Optional[bool],
+    out_file: str,
+    err_file: str,
+    telemetry_data_path: Path,
+) -> t.Dict[str, t.Any]:
+    return {
+        "name": model.name,
+        "path": model.path,
+        "run_settings": _dictify_run_settings(model.run_settings),
+        "batch_settings": _dictify_batch_settings(model.batch_settings)
+        if model.batch_settings
+        else None,
+        "params": model.params,
+        "files": {
+            "Symlink": model.files.link,
+            "Configure": model.files.tagged,
+            "Copy": model.files.copy,
+        }
+        if model.files
+        else {
+            "Symlink": [],
+            "Configure": [],
+            "Copy": [],
+        },
+        "colocated_db": {
+            "settings": model.run_settings.colocated_db_settings,
+            "scripts": [
+                {
+                    script.name: {
+                        "backend": "TORCH",
```

Comment:
should this always be torch?

Reply:
Right now, the only way a user can execute arbitrary python code on the database is if they compile it to torch script. So (for now) the `DBScript` does not have a `backend` attr and there is the (potentially naive) assumption throughout SmartSim that all `DBScript`s are using the Torch backend